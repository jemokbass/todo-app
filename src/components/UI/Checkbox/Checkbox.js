import React, { forwardRef } from 'react';
import PropTypes from 'prop-types';

import { ReactComponent as CheckSvg } from '@src/assets/img/check.svg';

const Checkbox = forwardRef((props, ref) => {
  const { checked, onChange, onBlur, disabled, name, title, onClick } = props;

  return (
    <div className="check">
      <div className="check-input">
        <input
          className="checkbox"
          type="checkbox"
          checked={checked}
          onChange={onChange}
          onBlur={onBlur}
          disabled={disabled}
          ref={ref}
          name={name}
          onClick={onClick}
        />
        <CheckSvg className="check-input__icon" />
      </div>
      <p className="check__title">{title}</p>
    </div>
  );
});

Checkbox.defaultProps = {
  onChange: () => {},
  onBlur: () => {},
  disabled: false,
  title: '',
  name: '',
};

Checkbox.propTypes = {
  title: PropTypes.string,
  onChange: PropTypes.func,
  onBlur: PropTypes.func,
  checked: PropTypes.bool,
  disabled: PropTypes.bool,
  name: PropTypes.string,
};

export default Checkbox;
