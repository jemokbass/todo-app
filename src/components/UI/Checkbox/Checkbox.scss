.check {
  display: flex;

  &__title {
    margin: 0;
    padding: 5px 0 0 10px;
    font-size: 16px;
  }
}

.check-input {
  position: relative;
  flex: 0 0 24px;
  width: 24px;
  height: 24px;
  margin: 0 0 15px;
  background: $element-color-lightest;

  .checkbox {
    position: absolute;
    z-index: 2;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    display: block;
    width: 100%;
    height: 100%;
    line-height: 0;
    opacity: 0;
    cursor: pointer;

    &:checked + .check-input__icon {
      opacity: 1;
    }
  }

  .checkbox[disabled] {
    opacity: 0.6;
  }

  &__icon {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    width: 20px;
    height: 18px;
    opacity: 0;
  }
}
