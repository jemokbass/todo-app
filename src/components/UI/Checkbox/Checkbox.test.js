import React from 'react';
import { render, screen } from '@testing-library/react';

import Checkbox from './Checkbox';

describe('Checkbox component', () => {
  it('Should render Checkbox component', () => {
    render(<Checkbox />);
  });

  it('Should render checkbox with title', () => {
    render(<Checkbox title="title" />);
    expect(screen.getByText(/title/i)).toBeInTheDocument();
  });

  describe('Should render snapshots', () => {
    it('Should render empty snapshot checkbox', () => {
      const { asFragment } = render(<Checkbox />);
      expect(asFragment(<Checkbox />)).toMatchSnapshot();
    });
  });

  describe('Default props', () => {
    it('Should use default onChange', () => {
      const res = Checkbox.defaultProps.onClick;
      expect(res).toBe(undefined);
    });
  });

  it('Should use default disabled', () => {
    const res = Checkbox.defaultProps.disabled;
    expect(res).toBe(false);
  });

  it('Should use default name', () => {
    const res = Checkbox.defaultProps.name;
    expect(res).toBe('');
  });

  it('Should use default title', () => {
    const res = Checkbox.defaultProps.title;
    expect(res).toBe('');
  });
});
