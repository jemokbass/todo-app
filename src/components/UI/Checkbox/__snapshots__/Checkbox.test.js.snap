// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component Should render snapshots Should render empty snapshot checkbox 1`] = `
<DocumentFragment>
  <div
    class="check"
  >
    <div
      class="check-input"
    >
      <input
        class="checkbox"
        name=""
        type="checkbox"
      />
      <svg
        class="check-input__icon"
      >
        check.svg
      </svg>
    </div>
    <p
      class="check__title"
    />
  </div>
</DocumentFragment>
`;
